### ISSUE TITLE
Add 'Subscribe to our newsletter" checkbox on checkout_success ...

### ISSUE THREAD

**Issue by lat9** [lat9/one_page_checkout#388]
... when asking if a guest wants to 'convert' their account to a standard one.

Requested in [this](https://www.zen-cart.com/showthread.php?220781-One-Page-Checkout-Support-Thread&p=1398330#post1398330) Zen Cart support-thread posting.

**Comment by JeffMash** [comment]
In my [original ZC posting link](https://www.zen-cart.com/showthread.php?220781-One-Page-Checkout-Support-Thread&p=1398330#post1398330), my request is slightly more nuanced than what I see here.

What I'd like is to have the "Subscribe to Newsletter" checkbox on the actual **CHECKOUT page** (_not just the Checkout Success page, although that is good too._)

My goal is that every customer by default is going to sign up for the newsletter by simply submitting their order (and they will have to uncheck the box if they don't want it).   This is similar to what we have now, where every customer signs up by default when creating their account.........but in case people with OLD accounts want to sign up again, they now will have this option enabled by default BEFORE they submit the order, and can opt-out by unchecking the box.

This can be configurable, of course, for those who don't want this option to be enabled.  But for people like us, it would help generate a lot of leads for our newsletter list.

We obviously would like the option for guests to sign up when converting their account, but if we can also do the above, that is what the original request was all about.

**Comment by simon1066** [comment]
> My goal is that every customer by default is going to sign up for the newsletter by simply submitting their order (and they will have to uncheck the box if they don't want it).

That's probably ok for CCPA but this wouldn't be allowed for those customers that fall under GDPR (or similar) regulations where consent _"must be unambiguous and involve a clear affirmative action (an opt-in)"_ and where pre-ticked opt-in boxes are specifically banned.

**Comment by lat9** [comment]
Yeah, the goal of OPC is to mimic the base 3-page create-account/3-page-checkout operation.  In that vein, adding the subscribe checkbox on the `checkout_success` page for guest-to-account conversions makes sense.  What you're describing is "custom coding".

**Comment by lat9** [comment]
> In my [original ZC posting link](https://www.zen-cart.com/showthread.php?220781-One-Page-Checkout-Support-Thread&p=1398330#post1398330), my request is slightly more nuanced than what I see here.
> 
> What I'd like is to have the "Subscribe to Newsletter" checkbox on the actual **CHECKOUT page** (_not just the Checkout Success page, although that is good too._)
> 
> My goal is that every customer by default is going to sign up for the newsletter by simply submitting their order (and they will have to uncheck the box if they don't want it). This is similar to what we have now, where every customer signs up by default when creating their account.........but in case people with OLD accounts want to sign up again, they now will have this option enabled by default BEFORE they submit the order, and can opt-out by unchecking the box.
> 
> This can be configurable, of course, for those who don't want this option to be enabled. But for people like us, it would help generate a lot of leads for our newsletter list.
> 
> We obviously would like the option for guests to sign up when converting their account, but if we can also do the above, that is what the original request was all about.

I see the functionality you're striving for, but the newsletter processing in Zen Cart applies to *customers*, not *guests*.  There's no capture (other than in an order itself) of a guest-customer's email address.
